crates/biome_service/src/diagnostics.rs

```diff
@@ -341,6 +343,30 @@ pub enum SearchError {
 )]
 pub struct InvalidPattern;
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct QueryError(pub String);
```

Comment:
nit: `QueryDiagnostic`

Reply:
Addressed in https://github.com/biomejs/biome/pull/3288/commits/240455ba282b4061510af5821e1c0bc3ee27ed32.